### PR TITLE
[Modules] Updated Authorization/PolicyExemptions to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Authorization/PolicyExemptions to new dependency approach

| Pipeline |
| - |
| [Authorization: PolicyExemptions](https://github.com/Azure/ResourceModules/actions/workflows/ms.authorization.policyexemptions.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Authorization/policyExemptions/.test/mg/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,45 @@
+targetScope = 'managementGroup'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'apemg'
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/policyExemptions/.test/mg/deploy.test.bicep:10]:
```suggestion
param serviceShort string = 'apemgcom'
```

File: modules/Microsoft.Authorization/policyExemptions/.test/rg/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'ms.authorization.policyexemptions-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'aperg'
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/policyExemptions/.test/rg/deploy.test.bicep:14]:
```suggestion
param serviceShort string = 'apergcom'
```

File: modules/Microsoft.Authorization/policyExemptions/.test/sub/deploy.test.bicep [hunk 3]
```diff
@@ -0,0 +1,44 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'apesub'
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/policyExemptions/.test/sub/deploy.test.bicep:10]:
```suggestion
param serviceShort string = 'apesubcom'
```

**Review by ahmadabdalla — COMMENTED** [review]
Thanks Alex. Minor suggestions on the service short name. And please change the folder names to use the `<scope>.common` method

File: modules/Microsoft.Authorization/policyExemptions/.test/mg/deploy.test.bicep (hunk 1 above)
Comment (MrMCake) [reply modules/Microsoft.Authorization/policyExemptions/.test/mg/deploy.test.bicep]: done

File: modules/Microsoft.Authorization/policyExemptions/.test/rg/deploy.test.bicep (hunk 2 above)
Comment (MrMCake) [reply modules/Microsoft.Authorization/policyExemptions/.test/rg/deploy.test.bicep]: done

File: modules/Microsoft.Authorization/policyExemptions/.test/sub/deploy.test.bicep (hunk 3 above)
Comment (MrMCake) [reply modules/Microsoft.Authorization/policyExemptions/.test/sub/deploy.test.bicep]: done

**Review by ahmadabdalla — APPROVED** [review]

**Pull request merged**